extensions/workloads/template.go

```diff
@@ -78,6 +86,14 @@ func NewDaemonSetTemplate(daemonsetName string, namespace string, template corev
 		template.ObjectMeta.Labels["workload.user.cattle.io/workloadselector"] = fmt.Sprintf("apps.daemonset-%v-%v", namespace, daemonsetName)
 	}
 
+	if isWindows {
```

Comment:
Yeah let's add this in NewPodTemplate that way it's a single if statement rather than affecting two functions for this.

Reply:
Done.

Reply:
I actually need this here. This specifically is causing upgrade tests to fail without it. It's because Linux and Windows worker nodes are being registered, so there needs to be the filter to properly have this.

Reply:
Can you just move this if statement to the NewPodTemplate?

Reply:
I can try tomorrow. I actually used a wrong label when I was doing local testing lol. Will follow up tomorrow.

Reply:
That worked, I did a quick local test with snapshots / upgrades and validated all was good.